components/UploadForm.vue

```diff
@@ -347,12 +401,29 @@ const processEPub = async ({ buffer, file }: { buffer: ArrayBuffer; file: File }
     }
     epubMetadataList.value.push(epubMetadata)
   } catch (err) {
+    // eslint-disable-next-line no-console
     console.error(err)
   }
 }
 
 const handleDeleteFile = (index: number) => {
-  fileRecords.value.splice(index, 1)
+  const [removedFile] = fileRecords.value.splice(index, 1)
+  if (removedFile.fileType?.startsWith('image/')) {
+    epubMetadataList.value = epubMetadataList.value
+      .map((metadata: any) => {
+        if (metadata.thumbnailIpfsHash === removedFile.ipfsHash && metadata.epubFileName) {
+          return { ...metadata, thumbnailIpfsHash: null, coverData: null }
+        }
+        return metadata
+      })
+      .filter((metadata: any) =>
+        metadata.epubFileName
+      )
```

Comment:
```suggestion
        if (metadata.thumbnailIpfsHash === removedFile.ipfsHash) {
          return { ...metadata, thumbnailIpfsHash: null, coverData: null }
        }
        return metadata
      })
      .filter((metadata: any) =>
        metadata.epubFileName  || metadata.thumbnailIpfsHash
      )
```

Just in case we have multiple images later?

Reply:
[🎨 Improve logic for deleting cover image](https://github.com/likecoin/nft-book-press/pull/354/commits/eccc195e05a965e004618f477d391ab696e8f7c9)